src/dev/i18n/serializers/json.js

```diff
@@ -19,10 +19,10 @@
 
 import { i18n } from '@kbn/i18n';
 
-export function serializeToJson(defaultMessages) {
-  const resultJsonObject = { formats: i18n.formats, messages: {} };
+export function serializeToJson(messages, formats = i18n.formats) {
+  const resultJsonObject = { formats, messages: {} };
 
-  for (const [mapKey, mapValue] of defaultMessages) {
+  for (const [mapKey, mapValue] of Array.isArray(messages) ? messages : Object.entries(messages)) {
```

Comment:
__issue:__ let's just accept only "array" format in this function as before and do `Object.entries` on the consumer side if needed.